### PR TITLE
fix(DATAGO-120206): Get agent card from remote agent

### PR DESCRIPTION
**Pull request by alimosaed**
### What is the purpose of this change?

1. **First:** A SAM application cannot retrieve an agent card of a remote agent that is connected via an A2A proxy from a customized path. It always uses the default agent card path, while the agent card path on the remote agent may differ.  
2. **Second:** If the remote agent requires authentication to retrieve an agent card, a SAM application fails to authenticate.

### How was this change implemented?

- For the first problem, the agent card path has been appended to the agent URL. This is accomplished by passing `agent_card_path` to the `A2ACardResolver` method. This method already supported a customized agent card path; however, the correct value was not being passed to it.

- For the second problem, synchronous authentication was added to the base class of the A2A proxy. This authentication is necessary to retrieve the agent card immediately when the proxy starts.

### Key Design Decisions

Both synchronous and asynchronous authentication methods are required because the proxy has two distinct discovery phases operating in different execution contexts. The base component’s `_perform_initial_discovery()` runs synchronously during `__init__()` to quickly populate the agent registry at startup using `httpx.Client()`, while the A2A component’s `_fetch_agent_card()` runs asynchronously on a recurring timer using `httpx.AsyncClient()` to refresh agent cards and support OAuth2 authentication (which requires awaiting token retrieval). Since async functions cannot be called from synchronous code without significant architectural changes, `build_static_auth_headers()` is used for the synchronous startup path (supporting only bearer/API key authentication), and `build_auth_headers_async()` is used for the asynchronous runtime path (supporting all authentication types, including OAuth2). Both methods share the same core authentication logic to avoid code duplication while respecting the fundamental synchronous/asynchronous boundary imposed by Python’s execution model.

### How was this change tested?

- [x] Manual testing: [describe scenarios]
- [x] Unit tests: [new/modified tests]
- [ ] Integration tests: [if applicable]
- [ ] Known limitations: [what wasn't tested]

### Is there anything the reviewers should focus on/be aware of?
Does the design make sense and if there is any edge cases.

### CODE REVIEW

File: src/solace_agent_mesh/common/auth_headers.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        for header_config in custom_headers_list:
+            header_name = header_config.get("name")
+            header_value = header_config.get("value")
+            if header_name and header_value:
+                headers[header_name] = header_value
+
+    return headers
+
+
+async def build_auth_headers_async(
```
Comment (enavitan) [inline src/solace_agent_mesh/common/auth_headers.py]: this method implies resemblance to the one above except its async - however in reality:
 - uses the static one to  build static headers
 - performs async calls to the oauth_toke_fetcher
 - pulls custom headers from the config
 - aggregates the results of all of them
 
So in short name doesn't reflect what exactly method does

this does feel more like some sort of a builder pattern however caller must be in charge of deciding what to pass in for final aggregation


```
HeaderBuilder
  .with(static_headers) <-- call to the static headers function 
  .with(await fetch_oauth_headers(...)) <-- call to oauth2 Auth headers
  .with(custom_headers) <-- call to the custom headers method 
  .build()
```
so in short - aggregator knows how to assemble these headers, while consumer decides what kind of header he interested in, WDYT?

File: src/solace_agent_mesh/agent/proxies/base/component.py [hunk 2]
```diff
@@ -420,8 +428,49 @@ def _initial_discovery_sync(self):
...
+
+                            if auth_type in ("oauth2_client_credentials", "oauth2_authorization_code"):
+                                # Skip this agent - OAuth2 not supported in sync context
+                                log.info(
+                                    "%s Skipping agent '%s' in synchronous discovery (OAuth2 configured). "
+                                    "Will be discovered during periodic async discovery.",
+                                    self.log_identifier,
+                                    agent_alias,
+                                )
+                                continue
```
Comment (enavitan) [inline src/solace_agent_mesh/agent/proxies/base/component.py:453]: lets improve flow a little:
 - we build static headers but later on  we might not even fetch anything if oauth_type is of certain type- thus we continue.

how about instead:
check if auth_type is of the type we need and if o so go ahead with static headers calcualtion, otherwise continue - not spend time building static_headers?

WDYT?

File: src/solace_agent_mesh/common/auth_headers.py (hunk 1 above)
Comment (alimosaed) [reply src/solace_agent_mesh/common/auth_headers.py]: Renamed the methods.

About the builder pattern, since we need only two scenarios of building a header, the simplest way is to have two separate methods. The builder pattern is a good option for scenarios when various scenarios can be generated using small methods.
So, definitely we can use the builder method but it does not add enough value. On the other hand, we have to test different authentication scenarios.
